api/src/main/java/jakarta/enterprise/inject/spi/CDI.java

```diff
@@ -79,6 +80,10 @@ private static CDIProvider getCDIProvider() {
             }
         } catch (IllegalStateException ignored) {
         }
+        //configuredProvider cannot be null if providerSetManually is true. If it somehow is an NPE is appropriate.
+        if (providerSetManually && configuredProvider.getCDI() == null) {
```

Comment:
You could add this logic directly into the `catch` block and just check for `providetSetManually`.

Reply:
Good idea!